js/utils/metrics.js

```diff
@@ -171,7 +171,7 @@ export function endAjaxEvent(key, segmentation) {
  */
 export function recordPrefixedEvent(eventName, prefix = '', segmentation = {}) {
   if (!eventName) throw new Error('Please provide an eventName');
-  if (segmentation && !(typeof segmentation === 'object')) {
+  if (segmentation && typeof segmentation !== 'object') {
```

Comment:
Actually, I don't even know why the first part is needed, this should be fine:

`if (typeof segmentation !== 'object') {`

and if null would bomb anything (e.g. if countly doesn't handle it), then you were right that you would need the null check.

`if (typeof segmentation !== 'object' || segmentation === null) {`